library/general/src/lib/yast2/popup.rb

```diff
@@ -0,0 +1,318 @@
+      #   If empty string is passed no headline is shown.
+      # @yield [] operation that needs to show feedback
+      # @return result of block
+      def feedback(message, headline: "", &block)
+        headline = generate_headline(headline)
+        if !message.is_a?(::String)
+          raise ArgumentError, "Invalid value #{message.inspect} of parameter message"
+        end
+
+        body = VBox(
+          VSpacing(0.4),
+          *headline_widgets(headline),
+          Left(Label(message))
+        )
+        content_res = content(body, {})
+
+        res = Yast::UI.OpenDialog(content_res)
+        raise "Failed to open dialog, see logs." unless res
+        begin
+          block.call
+        ensure
+          Yast::UI.CloseDialog
+        end
+      end
+
+    private
+
+      include Yast::UIShortcuts
+
+      def check_arguments!(message, details, timeout, focus, buttons)
+        if !message.is_a?(::String)
+          raise ArgumentError, "Invalid value #{message.inspect} of parameter message"
+        end
+
+        if !details.is_a?(::String)
+          raise ArgumentError, "Invalid value #{details.inspect} of parameter details"
+        end
+
+        if !timeout.is_a?(::Integer)
+          raise ArgumentError, "Invalid value #{timeout.inspect} of parameter timeout"
+        end
+
+        if !buttons.key?(focus)
+          raise ArgumentError, "Invalid value #{focus.inspect} for parameter focus. " \
+           "Known buttons: #{buttons.keys}."
+        end
+
+        nil
+      end
+
+      def generate_buttons(buttons)
+        case buttons
+        when ::Hash
+          buttons
+        when :ok
+          { ok: Yast::Label.OKButton }
+        when :continue_cancel
+          { continue: Yast::Label.ContinueButton, cancel: Yast::Label.CancelButton }
+        when :yes_no
+          { yes: Yast::Label.YesButton, no: Yast::Label.NoButton }
+        else
+          raise ArgumentError, "Invalid value #{buttons.inspect} for parameter buttons."
+        end
+      end
+
+      def generate_headline(headline)
+        case headline
+        when ::String
+          headline
+        when :warning
+          Yast::Label.WarningMsg
+        when :error
+          Yast::Label.ErrorMsg
+        else
+          raise ArgumentError, "Invalid value #{headline.inspect} for parameter headline."
+        end
+      end
+
+      def add_details_button(buttons)
+        # use this way merge to have details as first place button
+        { details: _("&Details...") }.merge(buttons)
+      end
+
+      def add_stop_button(buttons)
+        # use this way merge to have details as first place button
+        buttons.merge(stop: Yast::Label.StopButton)
+      end
+
+      def headline_widgets(headline)
+        if headline.empty?
+          [Empty()]
+        else
+          [Left(Heading(headline)), VSpacing(0.2)]
+        end
+      end
+
+      def timeout_widget(timeout)
+        if timeout > 0
+          Label(Id(:label), timeout.to_s)
```

Comment:
NP: maybe something like `:timeout_label` would be better here, `:label` is too generic IMHO...

Reply:
right, I keep same ID as old Popup and it use both, so longer will be better.